src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/CleanUpRejectedFilesTask.java

```diff
@@ -53,58 +54,60 @@ public void run() {
+
+        var containerName = containerClient.getBlobContainerName();
+        log.info("Looking for rejected files to delete. Container: {}", containerName);
+
+        containerClient
+            .listBlobs(listOptions, null)
+            .stream()
+            .filter(this::canBeDeleted)
+            .map(blobItem -> containerClient.getBlobClient(blobItem.getName()))
+            .forEach(blobClient -> leaseAcquirer.ifAcquiredOrElse(
+                blobClient,
+                leaseId -> delete(blobClient, leaseId),
+                errorCode -> {}, // nothing to do if blob not found in rejected container
+                false
+            ));
+
+        log.info("Finished removing rejected files. Container: {}", containerName);
+    }
+
+    private void delete(BlobClient blobClient, String leaseId) {
         try {
-            CloudBlockBlob blob = container.getBlockBlobReference(
-                zipFileName
+            blobClient.deleteWithResponse(
+                DeleteSnapshotsOptionType.INCLUDE,
+                new BlobRequestConditions().setLeaseId(leaseId),
+                null,
+                Context.NONE
             );
-
-            if (canBeDeleted(blob)) {
-                Optional<String> leaseId = blobManager.acquireLease(
-                    blob,
-                    container.getName(),
-                    zipFileName
-                );
-                if (leaseId.isPresent()) {
-                    AccessCondition accessCondition = generateLeaseCondition(leaseId.get());
-                    blob.delete(INCLUDE_SNAPSHOTS, accessCondition, null, null);
-                    log.info(
-                        "Deleted rejected file {} from container {}",
-                        blob.getName(),
-                        container.getName()
-                    );
-                }
-            }
-        } catch (URISyntaxException | StorageException exc) {
+            log.info(
+                "Deleted rejected file {} from container {}",
+                blobClient.getBlobName(),
+                blobClient.getContainerName()
+            );
+        } catch (Exception ex) {
             log.error(
                 "Unable to delete rejected file {} from container {}",
-                zipFileName,
-                container.getName(),
-                exc
+                blobClient.getBlobName(),
+                blobClient.getContainerName(),
+                ex
             );
         }
     }
 
-    private boolean canBeDeleted(CloudBlockBlob blob) throws StorageException {
-        blob.downloadAttributes();
-
-        Date createdTime = blob.getProperties().getLastModified();
-        Date cutoff = Date.from(Instant.now().minus(this.ttl));
-
-        return createdTime.before(cutoff);
+    private boolean canBeDeleted(BlobItem blobItem) {
+        return blobItem
+            .getProperties()
+            .getLastModified()
+            .plus(ttl)
+            .isBefore(now());
```

Comment:
this is offset now. it'll use whatever servers have configured. should we include timezone then? this also assuming `getLastModified()` is of same type

Reply:
I thought it should be same. I guess last modified and now()  both using the default.

Reply:
same code with blob-router,  but seems better to add zone UTC.

Reply:
Lodnon zone used.